File: Sorting algorithms/insertion.py
def insertion(arr):
    for i in range(len(arr)):
        j=i
        while j>0:
            if arr[j]<arr[j-1]: arr[j],arr[j-1]=arr[j-1],arr[j]
            else: break
            j-=1
    return arr

arr=[5,3,7,9,3,6,67,23,12,6,421,1,0]
print(insertion(arr))

File: Sorting algorithms/insertion.js
function insertion(arr,l,h){
  for(i=l;i<=h;i++){
    for(j=i;h>=0;j--){
      if(arr[j]<arr[j-1]){
        [[arr[j]],[arr[j-1]]]=[[arr[j-1]],[arr[j]]];
      }
      else {
        break;
      }
    }
  }
  return arr;
}

var arr = [456,23,888,123,34,33,1,0,45,99,320,50];
var l=0;
var h = arr.length-1;
console.log("Here is unsort array: "+arr);
console.log("Here is sorted array: "+insertion(arr,l,h));
